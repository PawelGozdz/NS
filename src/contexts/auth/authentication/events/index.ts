export * from './on-user-updated/on-user-updated.handler';
