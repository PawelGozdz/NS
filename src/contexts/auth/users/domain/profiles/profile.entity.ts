import { Address, EntityId } from '@libs/common';

import { ProfileSnapshot } from './profile.snapshot';

export type IProfileCreateData = {
	id: EntityId;
	userId: EntityId;
	firstName?: string;
	lastName?: string;
	username?: string;
	address?: Address;
	bio?: string;
	dateOfBirth?: Date;
	gender?: string;
	hobbies?: string[];
	languages?: string[];
	phoneNumber?: string;
	profilePicture?: string;
	rodoAcceptanceDate?: Date;
};

export type IProfileUpdateData = {
	firstName?: string | null;
	lastName?: string | null;
	username?: string | null;
	address?: Address | null;
	bio?: string | null;
	dateOfBirth?: Date | null;
	gender?: string | null;
	hobbies?: string[];
	languages?: string[];
	phoneNumber?: string | null;
	profilePicture?: string | null;
	rodoAcceptanceDate?: Date | null;
};

export class Profile {
	id: EntityId;

	userId: EntityId;

	firstName: string | null;

	lastName: string | null;

	username: string | null;

	address: Address | null;

	bio: string | null;

	dateOfBirth: Date | null;

	gender: string | null;

	hobbies: string[];

	languages: string[];

	phoneNumber: string | null;

	profilePicture: string | null;

	rodoAcceptanceDate: Date | null;

	constructor(props: IProfileCreateData) {
		this.id = props.id;
		this.userId = props.userId ?? null;
		this.firstName = props.firstName ?? null;
		this.lastName = props.lastName ?? null;
		this.username = props.username ?? null;
		this.address = props.address ?? null;
		this.bio = props.bio ?? null;
		this.dateOfBirth = props.dateOfBirth ?? null;
		this.gender = props.gender ?? null;
		this.hobbies = props.hobbies ?? [];
		this.languages = props.languages ?? [];
		this.phoneNumber = props.phoneNumber ?? null;
		this.profilePicture = props.profilePicture ?? null;
		this.rodoAcceptanceDate = props.rodoAcceptanceDate ?? null;
	}

	public static create(profile: IProfileCreateData) {
		return new Profile({
			id: profile.id,
			userId: profile.userId,
			firstName: profile.firstName,
			lastName: profile.lastName,
			username: profile.username,
			address: profile.address,
			bio: profile.bio,
			dateOfBirth: profile.dateOfBirth,
			gender: profile.gender,
			hobbies: profile.hobbies,
			languages: profile.languages,
			phoneNumber: profile.phoneNumber,
			profilePicture: profile.profilePicture,
			rodoAcceptanceDate: profile.rodoAcceptanceDate,
		});
	}

	public static restoreFromSnapshot(dao: ProfileSnapshot) {
		return new Profile({
			id: new EntityId(dao.id),
			userId: new EntityId(dao.userId),
			firstName: dao.firstName ?? undefined,
			lastName: dao.lastName ?? undefined,
			username: dao.username ?? undefined,
			dateOfBirth: dao.dateOfBirth ?? undefined,
			address: dao.address ? Address.create(dao.address) : undefined,
			bio: dao.bio ?? undefined,
			gender: dao.gender ?? undefined,
			hobbies: dao.hobbies,
			languages: dao.languages,
			phoneNumber: dao.phoneNumber ?? undefined,
			profilePicture: dao.profilePicture ?? undefined,
			rodoAcceptanceDate: dao.rodoAcceptanceDate ?? undefined,
		});
	}
}
