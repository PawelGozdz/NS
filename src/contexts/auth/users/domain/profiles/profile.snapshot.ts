import { Address } from '@libs/common';

export type ProfileSnapshot = {
	id: string;
	userId: string;
	firstName: string | null;
	lastName: string | null;
	username: string | null;
	address: Address | null;
	bio: string | null;
	gender: string | null;
	dateOfBirth: Date | null;
	hobbies: string[];
	languages: string[];
	phoneNumber: string | null;
	profilePicture: string | null;
	rodoAcceptanceDate: Date | null;
};
