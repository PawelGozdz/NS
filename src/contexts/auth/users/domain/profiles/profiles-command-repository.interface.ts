import { EntityId } from '@libs/common';

import { Profile } from './profile.entity';

export abstract class IProfilesCommandRepository {
	abstract save(user: Profile): Promise<void>;
	abstract getOneById(id: EntityId): Promise<Profile | undefined>;
	abstract getOneByUserId(email: string): Promise<Profile | undefined>;
}
