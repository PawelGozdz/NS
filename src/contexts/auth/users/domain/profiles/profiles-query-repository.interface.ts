import { CountryCode, EntityId } from '@libs/common';

export type ProfileInfo = {
	id: string;

	userId: string;

	firstName: string | null;

	lastName: string | null;

	dateOfBirth: Date | null;

	username: string | null;

	phoneNumber: string | null;

	gender: string | null;

	bio: string | null;

	hobbies: string[];

	languages: string[];

	profilePicture?: string | null;

	rodoAcceptanceDate?: Date | null;

	address: {
		street: string;
		streetNumber: string | undefined;
		city: string;
		countryCode: CountryCode;
		postalCode: string;
	} | null;
};

export abstract class IUsersQueryRepository {
	abstract getOneById(id: EntityId): Promise<ProfileInfo | undefined>;
	abstract getOneByUserId(userId: EntityId): Promise<ProfileInfo | undefined>;
}
