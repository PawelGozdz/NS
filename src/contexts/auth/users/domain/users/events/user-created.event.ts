import { Address, EntityId } from '@libs/common';
import { Event } from '@libs/cqrs';

export class UserCreatedEvent extends Event<UserCreatedEvent> {
	id: EntityId;
	email: string;
	profile: {
		id: EntityId;
		userId: EntityId;
		firstName: string | null;
		lastName: string | null;
		username: string | null;
		address: Address | null;
		bio: string | null;
		dateOfBirth: Date | null;
		gender: string | null;
		hobbies: string[];
		languages: string[];
		phoneNumber: string | null;
		profilePicture: string | null;
		rodoAcceptanceDate: Date | null;
	};

	constructor(event: UserCreatedEvent) {
		super(event);

		Object.assign(this, event);
	}
}
