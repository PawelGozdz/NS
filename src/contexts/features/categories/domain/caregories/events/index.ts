export * from './category-created.event';
export * from './category-updated.event';
