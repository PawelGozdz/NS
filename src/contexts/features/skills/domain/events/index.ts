export * from './skill-created.event';
