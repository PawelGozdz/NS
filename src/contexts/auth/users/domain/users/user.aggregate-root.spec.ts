import { EntityId } from '@libs/common';

import { Profile } from '../profiles';
import { UserCreatedEvent, UserUpdatedEvent } from './events';
import { User } from './user.aggregate-root';
import { UserAggregateRootFixtureFactory } from './user.aggregate-root.fixture';

const email = 'test@test.pl';

describe('User', () => {
  afterEach(() => {
    jest.resetModules();
    jest.restoreAllMocks();
  });

  describe('create', () => {
    describe('Success', () => {
      it('should apply UserCreatedEvent with property values', () => {
        // arrange

        // act
        const user = User.create({
          email,
          profile: {
            id: EntityId.createRandom(),
            userId: EntityId.createRandom(),
            firstName: 'test',
          },
        });

        // assert
        const events = user.getUncommittedEvents();

        expect(events.length).toEqual(1);
        expect(events[0]).toBeInstanceOf(UserCreatedEvent);

        expect(events[0]).toMatchSnapshot({
          id: expect.any(EntityId),
          profile: new Profile({
            firstName: 'test',
            id: expect.any(EntityId),
            userId: expect.any(EntityId),
          }),
        });
      });
    });
  });

  describe('update', () => {
    it('should apply UserUpdatedEvent with properties to be updated', () => {
      // arrange
      const user = UserAggregateRootFixtureFactory.create();

      const newEmail = 'new@test.pl';

      // act
      user.update({ email: newEmail });

      // assert
      const events = user.getUncommittedEvents();

      expect(events[0]).toBeInstanceOf(UserUpdatedEvent);
    });
  });
});
