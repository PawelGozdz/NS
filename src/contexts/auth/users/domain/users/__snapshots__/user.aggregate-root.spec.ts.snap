// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User create Success should apply UserCreatedEvent with property values 1`] = `
{
  "email": "test@test.pl",
  "id": Any<EntityId>,
  "profile": {
    "address": null,
    "bio": null,
    "dateOfBirth": null,
    "firstName": "test",
    "gender": null,
    "hobbies": [],
    "id": Any<EntityId>,
    "languages": [],
    "lastName": null,
    "phoneNumber": null,
    "profilePicture": null,
    "rodoAcceptanceDate": null,
    "userId": Any<EntityId>,
    "username": null,
  },
}
`;
